gapis/api/vulkan/replay.go

```diff
@@ -142,12 +143,30 @@ func (a API) Replay(
 	transforms = append(transforms, newDestroyResourcesAtEOS())
 	transforms = appendLogTransforms(ctx, replayType, c, transforms)
 
+	loopStart := numOfInitialCmds
+	loopEnd := api.CmdID(len(initialCmds) +len(c.Commands) -1)
 	cmdGenerator := commandGenerator.NewLinearCommandGenerator(initialCmds, c.Commands)
-	chain := transform.CreateTransformChain(ctx, cmdGenerator, transforms, out)
-	controlFlow := controlFlowGenerator.NewLinearControlFlowGenerator(chain)
-	if err := controlFlow.TransformAll(ctx); err != nil {
-		log.E(ctx, "%v Error: %v", replayType, err)
-		return err
+
+	_, _ = loopStart, loopEnd
+	switch firstRequest.(type) {
+       // case issuesRequest:
+       // case framebufferRequest:
+       case profileRequest:
+       		   log.W(ctx, "PERFORMANCE PERFORMANCE PERFORMANCE")
+               nullWriterObj := nullWriter{state: cloneState(ctx, c, out.State())}
+               chain := transform.CreateTransformChain(ctx, cmdGenerator, transforms, nullWriterObj)
+               controlFlow := NewLoopingVulkanControlFlowGenerator(ctx, chain, out, c, loopStart, loopEnd, 10)
```

Comment:
This two different initalisation can be reduced to  a function that returns you a control flow based on if you should frame loop or not. This approach would easily pollute the replay function with one or two more cases.